src-tauri/src/state/user_prompt.rs

```diff
@@ -1,104 +1,84 @@
+use oid4vci::credential_offer::CredentialOffer;
+use serde::{Deserialize, Serialize};
+use ts_rs::TS;
+
 /// "User prompts" are a way for the backend to communicate a desired/required user interaction to the frontend.
 /// This application design leaves it up to the frontend how it wants to handle such "user prompts".
 /// Having too much frontend logic in the backend would pollute the loose coupling and make it a lot harder to maintain.
 /// For example, the backend can "ask" the frontend to redirect the user to a specific page (e.g. the welcome page).
 /// Another example would be that the backend requires some user input to continue, but does not want to be specific about what
 /// is displayed to the user exactly.
-use serde::{Deserialize, Serialize};
-use ts_rs::TS;
-
-#[derive(Clone, Serialize, Deserialize, Debug, TS, PartialEq)]
-#[ts(export, export_to = "bindings/user-prompt/Redirect.ts")]
-#[serde(deny_unknown_fields)]
-pub struct Redirect {
-    pub r#type: CurrentUserPromptType,
-    pub target: String,
-}
-
-#[derive(Clone, Serialize, Deserialize, Debug, TS, PartialEq)]
-#[ts(export, export_to = "bindings/user-prompt/Warning.ts")]
-#[serde(deny_unknown_fields)]
-pub struct Warning {
-    pub r#type: CurrentUserPromptType,
-    pub message: String,
-}
-
-#[derive(Clone, Serialize, Deserialize, Debug, TS, PartialEq)]
-#[ts(export, export_to = "bindings/user-prompt/Selection.ts")]
-#[serde(deny_unknown_fields)]
-pub struct Selection {
-    pub r#type: CurrentUserPromptType,
-    /// An option is in the form: (<option_name>, <option_value>)
-    // pub options: Vec<(String, String)>,
-    pub options: Vec<String>,
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, TS)]
+#[serde(tag = "type")]
```

Comment:
Do we still need `#[serde(deny_unknown_fields)]` as well?

Reply:
added it now :+1: